### ISSUE TITLE
[Combobox] Mutating filter settings object doesn't work

### ISSUE THREAD

**Issue by montella1507** [telerik/kendo-angular#3425]
**Describe the bug**
https://www.telerik.com/kendo-angular-ui/components/dropdowns/combobox/filtering/

contains doesnt work.

https://user-images.githubusercontent.com/10418323/131698397-ab0a2954-5ba7-431a-a933-57b988131c27.mp4

**Comment by mbechev** [comment]
The example seems to be working if the **filterSettings** is replaced with new object:

```ts
  public changeFilterOperator(operator: 'startsWith' | 'contains'): void {
    this.filterSettings = { operator: operator };
  }
```

https://stackblitz.com/edit/angular-ixfcig

The demo needs to be updated as well as the type of the _operator_ parameter.

**Comment by montella1507** [comment]
Well.. thats propably because of immutability as default (it is input, right). However.. it doesnt work for me if it is default..  However.. i have older version so this may be the cause. 

    "@progress/kendo-angular-dropdowns": "4.2.4",

**Comment by dimitar-pechev** [comment]
Fixed in `@progress/kendo-angular-dropdowns@5.4.0`. [Here's a working example on our website.](https://www.telerik.com/kendo-angular-ui/components/dropdowns/combobox/filtering/#toc-built-in-directive)
